torchgeo/datasets/geo.py

```diff
@@ -329,7 +329,7 @@ def dtype(self) -> torch.dtype:
 
     def __init__(
         self,
-        root: str = "data",
+        root: Union[str, list[str]] = "data",
```

Comment:
According to [PEP-519](https://peps.python.org/pep-0519/#provide-specific-type-hinting-support), the correct way to support all possible path types is:
```python
Union[os.PathLike, str, bytes]
```
However, there are hundreds of other places in TorchGeo that would need to be modified to support pathlib. Let's stick with str for now and consider supporting pathlib in a separate issue/PR.

Reply:
Agree, let's keep str | list for now